libnmstate/state.py

```diff
@@ -123,6 +124,11 @@ def __init__(self, state):
             Route.KEY, {}).get(Route.CONFIG, [])
         self._config_iface_routes = State._index_routes_by_iface(
             self._config_routes)
+        dns_conf = self._state.get(DNS.KEY, {}).get(DNS.CONFIG, {})
+        self._config_dns = {
```

Comment:
I prefer you do not keep a separate unbounded dict for the DNS, just a pointer to the data in `self_state`.
If you change it, it is expected that the global one will change as well (otherwise they will get unsync).

See https://github.com/nmstate/nmstate/pull/387/commits/e03de943629a66a38e9855ba3601be1179bb1d8d#diff-09946053413e41502a142f4bfe77875bR419 for a suggested solution.